simpleflow/process/supervisor.py

```diff
@@ -80,22 +95,27 @@ def target(self):
             raise Exception("Child processes list is not empty, already called .start() ?")
 
         # start worker processes
-        for _ in range(self._nb_children):
-            child = multiprocessing.Process(
-                target=self._payload,
-                args=self._args
-            )
-            child.start()
-            self._processes.append(child)
+        self._start_worker_processes()
 
         # wait for all processes to finish
-        for proc in self._processes:
-            proc.join()
+        while True:
+            # if terminating, join all processes and exit the loop so we finish
+            # the supervisor process
+            if self._terminating:
+                for proc in self._processes:
+                    proc.join()
```

Comment:
Yes possible, if a process deadlocks it won't finish alone and we will be stuck here. But I'm not sure how to handle that and avoid breaking "normal" long tasks.